submariner/OVN-Interconnect.md

```diff
@@ -0,0 +1,231 @@
+driver. In the case of OVN it will be the IP of ovn-k8s-mp0 interface.
+
+The SubmarinerDefaultRoute CRD will also be created by Submariner Route-agent. It will have the list of remote clusters connected to the gateway.
+The nexthop will be the transit switch IP of the G/W node. If the transit switch IP is missing this CRD will not be created, which means it is
+a non-IC setup.
+
+The Routeagent will have these controllers added to it
+
+#### SubmarinerGWRouteCR Controller
+
+This controller will be responsible for programming the OVN cluster router, it will react only in a g/w node. When a SubmarinerGWRoute
+CR is created or modified the controller shall create or update a routing policy in OVN cluster router with a priority of 20000, and it should
+redirect any traffic destined to remote CIDR to the ovn-k8s-mp0 interface IP.
+
+```bash
+_uuid               : 0459f009-3603-47ac-8ee7-9d958540ed31
+bfd                 : []
+external_ids        : {}
+ip_prefix           : "10.132.0.0/16"
+nexthop             : "10.244.1.2"
+options             : {}
+output_port         : []
+policy              : []
+route_table         : ""
+```
+
+##### SubmarinerNonGWRouteController
+
+This controller will run in every in route agent. This controller connects to the OVN DB. When a SubmarinerDefaultRoute CR is created
+
+* non-gateway node : it updates the DB with a router policy with a priority 20000 to send the traffic to the remote cluster via next hop
+  mentioned, which is the transit switch IP to the g/w node. Before adding the route it checks if a route exists, if so it skips adding the
+  route again. This is required to prevent duplicate update since there can be more than one node in each zone and hence more than one
+  Routeagent.
+
+```bash
+_uuid               : 22db3005-64c5-4e32-aeb0-642423c30742
+action              : reroute
+external_ids        : {}
+match               : "ip4.dst==10.132.0.0/16"
+nexthop             : []
+nexthops            : ["169.254.0.1"]
+options             : {"external_ids:{submariner"="true}"}
+priority            : 20000
+```
+
+* gateway node/same zone node : In the gateway node route agent adds a route, if it does not exist, to send the traffic coming from other
+  zones which is destined to remote cluster IP range to the ovn-k8s-mp0 interface IP. It checks if the rule already exists to prevent duplicate
+  update.
+
+```bash
+_uuid               : d55185d8-3732-45c1-ae90-4a7f8cd191f7
+bfd                 : []
+external_ids        : {}
+ip_prefix           : "10.132.0.0/16"
+nexthop             : "10.244.1.2"
+options             : {}
+output_port         : []
+policy              : []
+route_table         : ""
+```
+
+### Backward Compatibility
+
+With this architecture, we are removing the logical switches and routers that Submariner creates. During migration, we will need to delete
```

Comment:
Also, it isn't just OVN version and IC being updated, but also new Submariner CRs. New datapath won't be available till submariner version upgrades and installs the new CRs, even if IC is already available.

Better to add a separate section on `Upgradability` that details how upgrades will be, or should be handled. Even if nothing to do in code, it should serve as base for documentation.

Reply:
This section can be revisited when we have a clear plan for upgrades.